### PR TITLE
Minor: re-order optimizer and scheduler steps

### PR DESCRIPTION
**Pull request by pyeres**
This PR swaps the order of optimizer and learning rate scheduler steps (addressing the warning pointed out by @HaokunLiu):

Warning message pre-change:
> UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`. Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate

### CODE REVIEW

**Comment by jeswan** [comment]
> Can we double check when/how Transformers fixed this?

Changed for SQuAD here: https://github.com/huggingface/transformers/pull/1136/files

**Comment by pyeres** [comment]
> Can we double check when/how Transformers fixed this?

A few re-orders were [done in 8/19](https://github.com/huggingface/transformers/search?q=scheduler.step%28%29&type=Commits), with "[One last reorder of {scheduler,optimizer}.step()](https://github.com/huggingface/transformers/commit/cf72479bf11bf7fbc499a518896dfd3cafdd0b21)" in 3/20.
